orttraining/orttraining/python/training/torchdynamo/ort_backend.py

```diff
@@ -16,18 +16,47 @@
 import torch.fx
 import torch.jit
 import torch.onnx
+import torch.onnx._internal
+import torch.onnx._internal.diagnostics
+import torch.onnx._internal.exporter
+import torch.onnx._internal.fx.decomposition_table
+import torch.onnx._internal.fx.passes
 import torch.onnx._onnx_supported_ops
-from torch._decomp import decomposition_table
 from torch._subclasses.fake_tensor import FakeTensor
 from torch.fx.passes.fake_tensor_prop import FakeTensorProp
 from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
 from torch.fx.passes.operator_support import OperatorSupport
 from torch.fx.passes.tools_common import CALLABLE_NODE_OPS
+
 from torch.onnx._globals import GLOBALS as ONNX_GLOBALS
 
 import onnxruntime  # type: ignore
 from onnxruntime.capi import _pybind_state as ORTC
 
+# DEFAULT_ONNX_EXPORTER_OPTIONS contains shared information between exporter and DORT.
+# For example, they should use the same decomposition table to maintain the same set
+# operators when
+#  1. capturing FX graph in torch.compile
+#  2. call exporter's API to convert `torch.fx.GraphModule` to ONNX model.
+DEFAULT_ONNX_EXPORTER_OPTIONS = torch.onnx._internal.exporter.ResolvedExportOptions(
+    torch.onnx._internal.exporter.ExportOptions()
```

Comment:
`ResolvedExportOptions` was meant to be the private (and consolidated version) of `ExportOptions`. Why not use `ExportOptions` directly?

We usually go with `ResolvedExportOptions` when we want to change a private attribute for debugging or accessing an invisible feature.

Maybe `onnx_registry` and `decomposition_table` should be public properties to prevent the `ResolvedExportOptions(ExportOptions())` pattern to hack the internals?

Reply:
Well, DORT is a kind of internal component of FX-to-ONNX exporter. I personally won't bother for this. :)